apps/server/src/modules/room/api/test/room-stats.api.spec.ts

```diff
@@ -0,0 +1,228 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { GroupEntityTypes } from '@modules/group/entity/group.entity';
+import { groupEntityFactory } from '@modules/group/testing';
+import { roomMembershipEntityFactory } from '@modules/room-membership/testing';
+import { RoomRolesTestFactory } from '@modules/room/testing/room-roles.test.factory';
+import { schoolEntityFactory } from '@modules/school/testing';
+import { ServerTestModule, serverConfig, type ServerConfig } from '@modules/server';
+import { HttpStatus, INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { cleanupCollections } from '@testing/cleanup-collections';
+import { UserAndAccountTestFactory } from '@testing/factory/user-and-account.test.factory';
+import { TestApiClient } from '@testing/test-api-client';
+import { roomEntityFactory } from '../../testing/room-entity.factory';
+import { SchoolEntity } from '@modules/school/repo';
+import { User } from '@modules/user/repo';
+import { AccountEntity } from '@modules/account/repo';
+import { schoolSystemOptionsEntityFactory } from '@modules/legacy-school/testing';
+import { SchoolSystemOptionsEntity } from '@modules/legacy-school/entity';
+
+describe('Room Controller (API)', () => {
+	let app: INestApplication;
+	let em: EntityManager;
+	let testApiClient: TestApiClient;
+	let config: ServerConfig;
+
+	beforeAll(async () => {
+		const moduleFixture = await Test.createTestingModule({
+			imports: [ServerTestModule],
+		}).compile();
+
+		app = moduleFixture.createNestApplication();
+		await app.init();
+		em = app.get(EntityManager);
+		testApiClient = new TestApiClient(app, 'rooms/stats');
+
+		config = serverConfig();
+	});
+
+	beforeEach(async () => {
+		await cleanupCollections(em);
+		config.FEATURE_ADMINISTRATE_ROOMS_ENABLED = true;
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	describe('GET /rooms/stats', () => {
+		describe('when the user is not authenticated', () => {
```

Comment:
do we also want to test if the feature is disabled? just a thought

Reply:
fixed: I added the test here to get the sonarcloud coverage for the line.